### DISCUSSION TITLE
LM should look more aggressively for WiFi networks

### DISCUSSION THREAD

**Discussion by kacperpaczos** [linuxmint/discussions#482] [Development]
I am using my phone to spread WiFi.

When I turn on the hotspot, I have to wait 1-2 minutes to connect to my laptop. I can speed this up by turning off WiFi through the applet and turning it on again. Then I have an instant connection.

Anyone, have the same?

I believe that in case we do not have a connection and we are in the first 5 minutes after logging in, we should increase the aggression of the WiFi applet.

Please describe your experiences to make this applet more user-friendly for everyone.

**Comment by Mihau13** [comment]
I have a similar problem. If the applet is kept open, even 10 minutes can pass since opening a hotspot without it becoming visible. Interestingly, after a while other networks which I'm not connected to, may become invisible without the target network appearing. Repeatedly opening and closing the applet can bring this down to a couple seconds but there's still a noticeable delay. Turning the WiFi off and on in the applet makes the target network appear immediately after fully starting up. Similar delays occur between a hotspot being closed and it disappearing from the applet.

With my workflow, I often have to switch between networks multiple times throughout the day, so intensifying searching only in the first five minutes after logging in and only if there's no connection, would not help me much. A solution that I think would work better, at least for me, would be adding a refresh button in the applet, that when clicked, forces the same search process that occurs when WiFi is first turned on.
